gnovm/pkg/gnolang/realm.go

```diff
@@ -509,7 +512,7 @@ func (rlm *Realm) decRefDeletedDescendants(store Store, oo Object) {
 		if rc == 0 {
 			rlm.decRefDeletedDescendants(store, child)
 		} else if rc > 0 {
-			// do nothing
+			rlm.MarkDirty(child)
```

Comment:
curious isn't it marked in previous phases?

Reply:
No, it isn't marked as dirty; we change the reference count, but it is never saved.